documentation/datastructure.dot

```diff
@@ -0,0 +1,210 @@
+
+    Attribution -> agent;
+    Attribution -> hasRole;
+    agent -> Person;
+    agent -> Organization;
+
+    DataDownload -> URL;
+
+    URL_ [label="URL"];
+    Place -> URL_;
+
+    streetAddress [label="streetAddress\nString"];
+    PostalAddress -> streetAddress
+    postalCode [label="postalCode\nString"];
+    PostalAddress -> postalCode
+    addressLocality [label="addressLocality\nString"];
+    PostalAddress -> addressLocality
+
+    // Enum
+    Types [color=brown, shape=record, label="{xml | text | image | movie | audio}"];
+
+
+    // Dataset
+    // -------
+    sameAs [label="sameAs\nURL"];
+    Dataset -> sameAs [label="0-n"];
+
+    hasTitle [label="hasTitle\nString"];
+    Dataset -> hasTitle [label="1"];
+
+    hasAlternativeTitle [label="hasAlternativeTitle\nString"];
+    Dataset -> hasAlternativeTitle [label="0-1"];
+
+    hasAbstract [label="hasAbstract\nString | URL"];
+    Dataset -> hasAbstract [label="1-n"];
+
+    Dataset -> hasTypeOfData [label="1-n"];
+    hasTypeOfData -> Types;
+
+    hasDocumentation [label="hasDocumentation\nString | URL"];
+    Dataset -> hasDocumentation [label="0-n"];
+
+    hasLicense [label="hasLicense\nURL"];
+    Dataset -> hasLicense [label="1-n"];
+
+    hasConditionsOfAccess [label="hasConditionsOfAccess\nString"];
+    Dataset -> hasConditionsOfAccess [label="1"];
+
+    hasHowToCite [label="hasHowToCite\nString"];
+    Dataset -> hasHowToCite [label="1"];
+
+    hasStatus [label="hasStatus\nString"];
+    Dataset -> hasStatus [label="1"];
+
+    hasDatePublished [label="hasDatePublished\nDate"];
+    Dataset -> hasDatePublished [label="0-1"];
+
+    hasLanguage [label="hasLanguage\nString"];
+    Dataset -> hasLanguage [label="1-n"];
+
+    Dataset -> isPartOf [label="1"];
+    isPartOf -> Project;
+
+    Dataset -> hasQualifiedAttribution [label="1-n"];
+    hasQualifiedAttribution -> Attribution;
+
+    hasDateCreated [label="hasDateCreated\nDate"];
+    Dataset -> hasDateCreated [label="0-1"];
+
+    hasDateModified [label="hasDateModified\nDate"];
+    Dataset -> hasDateModified [label="0-1"];
+
+    Dataset -> hasDistribution [label="0-1"];
+    hasDistribution -> DataDownload
+
+
+    // Project
+    // -------
+    hasName [label="hasName\nString"];
+    Project -> hasName [label="1"];
+
+    hasDescription [label="hasDescription\nString"];
+    Project -> hasDescription [label="1"];
+
+    hasKeywords [label="hasKeywords\nString"];
+    Project -> hasKeywords [label="1-n"];
+
+    hasDiscipline [label="hasDiscipline\nString | URL"];
+    Project -> hasDiscipline [label="1-n"];
+
+    hasStartDate [label="hasStartDate\nDate"];
+    Project -> hasStartDate [label="1"];
+
+    hasEndDate [label="hasEndDate\nDate"];
+    Project -> hasEndDate [label="0-1"];
+
+    hasTemporalCoverage [label="hasTemporalCoverage\nString | URL"];
+    Project -> hasTemporalCoverage [label="1-n"];
+
+    Project -> hasSpatialCoverage [label="1-n"];
+    hasSpatialCoverage -> Place
```

Comment:
Should arrays be also somehow better presented? Example: `hasSpatialCoverage\nPlace[]` Or just `[]`/`Array` since we type specified in next bubble.

Reply:
technically, that's not an array, because it only takes one Place (whereas there can be multiple hasSpatialCoverage)

but I can add some more types, though I'm not sure that helps a lot. But it's also not a lot of work...

Reply:
Take a look on the [example response](https://drive.google.com/file/d/17Xew80t8kg5AdWlnzkVN4w31UzNlo7aM/view) we have. It has many there is only one `hasSpatialCoverage` with Places inside. Is the example bad or ??

Reply:
The example response with `hasSpatialCoverage` looks good to me.

Reply:
I was thinking in turtle again. I guess the two are exactly equivalent, what we got is just more elegant JSON. :-)

Reply:
Yes but the output is the Place array, isn't it? Not like you mentioned 1-n `hasSpatialCoverage` with `Place` object inside:
![image](https://user-images.githubusercontent.com/29251288/99634792-5c04ee80-2a41-11eb-9d3c-066fa9f6f960.png)

Reply:
unless my logic is wrong, the following should be equivalent (though not identical):
```JSON
[
  {
    "hasSpatialCoverage": {
      "place": "Place A"
    }
  },
  {
    "hasSpatialCoverage": {
      "place": "Place B"
    }
  }
]
```
And
```json
{
  "hasSpatialCoverage": [
    {
      "place": "Place A"
    },
    {
      "place": "Place B"
    }
  ]
}
```
So, without having tested it, I'd assume that the turtle output would correspond to the first case, while the json corresponds to the second. In that sense, both representations are equally correct and incorrect at the same time.  Or am I completely confused (which is a realistic option)?

Reply:
I'm also a bit confused bit that, however the second option is returned via the API, so I need to address mapping to that what I get. I'll try to test if the first option will also work, when I get the mapping working ;).

Reply:
at least we're confused together ;)